src/Simplex/Messaging/Agent/Protocol.hs

```diff
@@ -429,6 +430,10 @@ data ConnectionErrorType
     DUPLICATE
   | -- | connection is simplex, but operation requires another queue
     SIMPLEX
+  | -- | queue status is prohibited in this context
+    BAD_QUEUE_STATUS
+  | -- | no public key stored for queue activation
+    NO_PUBLIC_KEY
```

Comment:
I don't think either of these errors are needed - they are implementation rather than protocol error, so should be reported as INTERNAL

Reply:
done